pkg/nodeagent/controller/operatingsystemconfig/containerd.go

```diff
@@ -123,6 +128,86 @@ Environment="PATH=` + extensionsv1alpha1.ContainerDRuntimeContainersBinFolder +
 	return nil
 }
 
+// ensureContainerdConfiguration sets the configuration for containerd.
+func (r *Reconciler) ensureContainerdConfiguration() error {
+	config, err := r.FS.ReadFile(configFile)
+	if err != nil {
+		return fmt.Errorf("unable to read containerd config.toml: %w", err)
+	}
+
+	content := map[string]any{}
+
+	err = toml.Unmarshal(config, &content)
+	if err != nil {
+		return fmt.Errorf("unable to decode containerd default config: %w", err)
+	}
+
+	type (
+		patch struct {
+			name  string
+			path  structuredmap.Path
+			setFn structuredmap.SetFn
+		}
+	)
+
+	patches := []patch{
+		{
+			name: "registry config path",
+			path: structuredmap.Path{"plugins", "io.containerd.grpc.v1.cri", "registry", "config_path"},
+			setFn: func(_ any) (any, error) {
+				return certsDir, nil
+			},
+		},
+		{
+			name: "imports paths",
+			path: structuredmap.Path{"imports"},
+			setFn: func(value any) (any, error) {
+				importPath := path.Join(configDir, "*.toml")
+
+				imports, ok := value.([]any)
+				if !ok {
+					return []string{importPath}, nil
+				}
+
+				for _, imp := range imports {
+					path, ok := imp.(string)
+					if !ok {
+						continue
+					}
+
+					if path == importPath {
+						return value, nil
+					}
+				}
+
+				return append(imports, importPath), nil
+			},
+		},
+	}
+
+	for _, p := range patches {
+		content, err = structuredmap.SetMapEntry(content, p.path, p.setFn)
```

Comment:
Hm, I think importing an `internal` package doesn't work, does it?

Reply:
I don't recall the details why we didn't copy it during the Hackathon and instead wrote our own variant. Maybe because of dependencies?! @Gerrit91 can you tell?

Reply:
Hey. The code uses both the BurntSushi and the Pelletier TOML dependencies, and we thought we wanted to stick to a single dependency for config management. Since we were able to write the map patcher in a very short time with few lines of code, we thought it was a better solution than copying this. In the end, maintainers will have to decide what they prefer. Certainly both solutions are viable.